webui/react/src/pages/TrialDetails/TrialChart.tsx

```diff
@@ -74,11 +98,11 @@ const TrialChart: React.FC<Props> = ({
       legend: { show: false },
       plugins: [ tooltipsPlugin(), trackAxis() ],
       scales: {
-        x: { time: false },
-        y: { distr: scale === Scale.Log ? 3 : 1 },
+        x: xScale,
+        y: yScale,
       },
       series: [
-        { label: 'Batch' },
+        { label: `Trial ${trialId} Batch` },
```

Comment:
This is basically a hack to force a recreate, but IMO it's good QOL improvement if people are switching between trials a lot. Only shows up on hover so pretty unobtrusive.

Details on why resorted to this in comments on the `UPlotChart.tsx` component.